src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,89 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedHashMap;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String csvSplitBy = ",";
+    private static final int operationTypeIndex = 0;
+    private static final int operationAmountIndex = 1;
+    private static final String resultingRowName = "result";
+    private static final String buyRowName = "buy";
+    private static final String supplyRowName = "supply";
+    private LinkedHashMap<String, Integer> hashtable = new LinkedHashMap<>();
+    private List<String> linesFromCsvFile;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        fillHashTable(fromFileName);
+        calcResultRow();
+        writeStatToFile(toFileName);
+    }
+
+    public void writeStatToFile(String toFile) {
+        try (FileWriter clearFileWriter = new FileWriter(toFile, false)) {
+            clearFileWriter.write(""); // To clear file
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFile, true))) {
+            iterateThoughHashTableRecords(bufferedWriter);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void iterateThoughHashTableRecords(BufferedWriter bufferedWriter) throws IOException {
+        for (String key : hashtable.keySet()) {
+            int value = hashtable.get(key);
+            String data = formCsvLine(key, value);
+            bufferedWriter.write(data);
+            bufferedWriter.newLine();
+        }
+    }
+
+    public void fillHashTable(String fromFile) {
+        readLinesFromCsv(fromFile);
+        constructHashTable();
+        for (String cvsLine : linesFromCsvFile) {
+            Operation operation = splitCvs(cvsLine);
+            int previousAmountInCell = hashtable.get(operation.type());
+            previousAmountInCell += operation.amount();
+            hashtable.put(operation.type(), previousAmountInCell);
+        }
+    }
+
+    public void readLinesFromCsv(String csvFilePath) {
+        try {
+            linesFromCsvFile = Files.readAllLines(Path.of(csvFilePath));
+        } catch (IOException e) {
+            throw new RuntimeException("DevCap: Exception while reading from CVS file -- ", e);
+        }
+    }
```

Comment:
Return data in String from this method

Reply:
I can use string, but then i should once more split this string to iterate though it. Am I wrong somewhere?